rage/src/bin/rage/main.rs

```diff
@@ -131,9 +131,9 @@ fn read_recipients(
     identity_strings: Vec<String>,
     max_work_factor: Option<u8>,
 ) -> Result<Vec<Box<dyn Recipient + Send>>, error::EncryptError> {
-    let mut recipients: Vec<Box<dyn Recipient + Send>> = vec![];
-    let mut plugin_recipients: Vec<plugin::Recipient> = vec![];
-    let mut plugin_identities: Vec<plugin::Identity> = vec![];
+    let mut recipients: Vec<Box<dyn Recipient + Send>> = Vec::with_capacity(recipient_strings.len());
+    let mut plugin_recipients: Vec<plugin::Recipient> = Vec::with_capacity(recipient_strings.len());
+    let mut plugin_identities: Vec<plugin::Identity> = Vec::with_capacity(recipient_strings.len());
```

Comment:
These always over-allocate, so I don't see the benefit of this change over the existing heuristical over-allocations. Please remove this change from the PR.

Reply:
```suggestion
    let mut recipients: Vec<Box<dyn Recipient + Send>> = vec![];
    let mut plugin_recipients: Vec<plugin::Recipient> = vec![];
    let mut plugin_identities: Vec<plugin::Identity> = vec![];
```